app/src/main/java/me/hufman/androidautoidrive/carapp/music/views/PlaybackView.kt

```diff
@@ -386,11 +406,26 @@ class PlaybackView(val state: RHMIState, val controller: MusicController, val ca
 
 	private fun redrawSong() {
 		val song = controller.getMetadata()
-		artistModel.value = if (controller.isConnected()) {
+
+		val artistTitle = if (controller.isConnected()) {
 			UnicodeCleaner.clean(song?.artist ?: "")
 		} else { L.MUSIC_DISCONNECTED }
-		albumModel.value = UnicodeCleaner.clean(song?.album ?: "")
-		trackModel.value = UnicodeCleaner.clean(song?.title ?: "")
+		artistTextScroller = TextScroller(artistTitle, MUSIC_METADATA_MAX_LINE_LENGTH)
+		artistModel.value = artistTextScroller.getText()
+
+		val albumTitle = UnicodeCleaner.clean(song?.album ?: "")
+		albumTextScroller = TextScroller(albumTitle, MUSIC_METADATA_MAX_LINE_LENGTH)
+		albumModel.value = albumTextScroller.getText()
+
+		val trackTitle = UnicodeCleaner.clean(song?.title ?: "")
+		val trackMaxLineLength = if (state is RHMIState.AudioHmiState) {
+			AUDIOSTATE_PLAYLIST_MAX_LINE_LENGTH
+		} else {
+			MUSIC_METADATA_MAX_LINE_LENGTH
+		}
+		trackTextScroller = TextScroller(trackTitle, trackMaxLineLength)
+		trackModel.value = trackTextScroller.getText()
+
```

Comment:
You could replace these model updates with a call to redrawLongTitles

Reply:
Yeah that's a lot more consistent, updated